File: 16주차/DE/463. Island Perimeter.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
 function islandPerimeter(grid) {
    for(let i = 0; i < grid.length; i++){
        grid[i].unshift(0)
        grid[i].push(0)
    }
    grid.unshift(Array(grid[0].length).fill(0))
    grid.push(Array(grid[0].length).fill(0))
    
    let dir = [[-1, 0], [0, -1], [1, 0], [0, 1]]
    let cnt = 0
    for(let i = 1; i < grid.length - 1; i++){
        for(let j = 1; j <grid[0].length - 1; j++){
            if(grid[i][j] == 1){
                for(let d of dir){
                    if(grid[i+d[0]][j+d[1]] == 0){
                        cnt++
                    }
                }
            }
        }
    }
    return cnt
};

File: 16주차/KM/463-island-perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int answer = 0;
        int[] dx = {1, -1, 0, 0};
        int[] dy = {0, 0, 1, -1};
        for(int i = 0; i<grid.length; i++){
            for(int j = 0; j<grid[i].length; j++){
                if(grid[i][j] == 0) continue;
                int cnt = 0;
                for(int k = 0; k<4; k++){
                    int nx = i + dx[k];
                    int ny = j + dy[k];
                    if(isIn(grid, nx, ny)) continue;
                    if(grid[nx][ny] == 1) cnt++;
                }
                answer += 4 - cnt;
            }
        }
        return answer;
    }
    public boolean isIn(int[][] grid, int x, int y){
        return x < 0 || x >= grid.length || y < 0 || y >= grid[0].length;
    }
}